src/cmdhandler/command_handler.cpp

```diff
@@ -370,28 +440,218 @@ std::string command_handler::collector_system(const std::vector<std::string>& to
     }
 }
 
+/**
+ * Modifies variables in the 'CameraSystem' struct in 'Variables' class. <br>
+ * system functions (on variables mentioned above): set, get, list, and delete
+ *
+ * @param tokens tokenized vector version of user command
+ * @param current_state server's current state struct
+ * @return response to user command as string
+ */
+std::string command_handler::camera_system(const std::vector<std::string>& tokens, StateVariables& current_state){
+    if(tokens[0] == "list"){
+        std::stringstream response;
+        response << "Current camera variables:";
+
+        //add url variable
+        response << "\n    url: "+current_state.camera.url;
+
+        //add camera_matrix variable
+        response << "\n    camera_matrix: ";
+        cv::Mat camera_matrix = current_state.camera.camera_matrix;
+        for (int row = 0; row < camera_matrix.rows; ++row) {
+            for (int col = 0; col < camera_matrix.cols; ++col) {
+                response << camera_matrix.at<double>(row, col);
+                response << ",";
+            }
+        }
+
+        //add distortion_matrix variable
+        response << "\n    distortion_matrix: ";
+        cv::Mat distortion_matrix = current_state.camera.distortion_matrix;
+        for (int row = 0; row < distortion_matrix.rows; ++row) {
+            for (int col = 0; col < distortion_matrix.cols; ++col) {
+                response << distortion_matrix.at<double>(row, col);
+                response << ",";
+            }
+        }
+
+        //add marker_dictionary variable
+        response << "\n    marker_dictionary: ";
+        response << current_state.camera.marker_dictionary;
+
+        //add camera_option variable
+        response << "\n    camera_options: ";
+        for(auto const &option : current_state.camera.camera_options){
+            response << "\n        " << option.first << ": " << std::boolalpha << option.second;
+        }
+
+        return response.str();
+    }else if(tokens[0] == "set"){
+        if(tokens.size() < 3){
+            return "please provide a variable to set\n    ex: set camera url http://example.com";
+        }
+
+        std::string variable = tokens[2];
+
+        if(variable == "url"){
+            if(tokens.size() != 4){
+                return "please provide a value for variable '"+variable+"'\n    ex: set camera url http://example.com";
+            }
+            current_state.camera.url = tokens[3];
+            return "camera url set to '"+tokens[3]+"'";
+        }else if(variable == "camera_matrix" || variable == "distortion_matrix"){
+            // since camera_matrix/distortion_matrix are the same data type/format, just use a conditional to assign a
+            // cv::Mat to the chosen variable
+            if(tokens.size() != 4){
```

Comment:
The camera matrix and distortion coefficients require a different number of values (camera matrix is 9, dist. coeff. is 5) so there should be a separate error message for each